terminology/terminology_processor.bal

```diff
@@ -0,0 +1,1661 @@
+            );
+            }
+
+            if valueSet.'version == () {
+                return r4:createFHIRError(
+            string `Can not find the version of the ValueSet with name: ${valueSet.name.toBalString()}`,
+            r4:ERROR,
+            r4:INVALID_REQUIRED,
+            diagnostic = string `Add appropriate version for the resource: 
+                                     http://hl7.org/fhir/R4/valueset-definitions.html#ValueSet.version`,
+            errorType = r4:VALIDATION_ERROR,
+            httpStatusCode = http:STATUS_BAD_REQUEST
+            );
+            }
+
+            string url = <string>valueSet.url;
+            string rVersion = <string>valueSet.'version;
+            string rKey = string `${url}|${rVersion}`;
+
+            if self.finder is () && self.valueSets.hasKey(rKey) {
+                return r4:createFHIRError(
+            "Duplicate entry",
+            r4:ERROR,
+            r4:PROCESSING_DUPLICATE,
+            diagnostic = string `Already there is a ValueSet exists in the registry with the URL: ${url}`,
+            errorType = r4:VALIDATION_ERROR,
+            httpStatusCode = http:STATUS_BAD_REQUEST
+            );
+            }
+
+            r4:FHIRValidationError? validateResult = validator:validate(valueSet.clone(), i4:ValueSet);
+
+            if validateResult is r4:FHIRValidationError {
+                return r4:createFHIRError(
+            "Validation failed",
+            r4:ERROR,
+            r4:INVALID,
+            diagnostic = string `Check whether the data conforms to the specification: 
+                                    http://hl7.org/fhir/R4/valueset-definitions.html`,
+            errorType = r4:VALIDATION_ERROR,
+            cause = validateResult,
+            httpStatusCode = http:STATUS_BAD_REQUEST
+            );
+            }
+
+            if self.finder is Finder {
+                r4:FHIRError[]? result = (<Finder>self.finder).addValueSets([valueSet.clone()]);
+                if result is r4:FHIRError[] {
+                    return result[0];
+                }
+            } else {
+                self.valueSets[rKey] = valueSet.clone();
+            }
+        }
+    }
+
+    # Find a Code System based on the provided Id and version.
+    #
+    # + id - Id of the CodeSystem to be retrieved
+    # + version - Version of the CodeSystem to be retrieved and it should be provided with system parameter,
+    # if this version parameter is not supplied then the latest version of CodeSystem will picked up.
+    # + return - Return CodeSystem data if the request is successful, return FHIR error if no data found for the provided Id
+    public isolated function readCodeSystemById(string id, string? 'version = ()) returns i4:CodeSystem|r4:FHIRError {
+        lock {
+            if self.finder is Finder {
+                return (<Finder>self.finder).findCodeSystem(id = id, version = 'version);
+            } else {
```

Comment:
We can extract out the else block

Reply:
What I was saying is since we return in the `if block` we don't need an else block to hold the rest of the code. we can just write them after the `if block`

Reply:
Updated